src/sync_connection_wrapper.rs

```diff
@@ -0,0 +1,254 @@
+        tokio::task::spawn_blocking(move || C::establish(&database_url))
+            .await
+            .unwrap_or_else(|e| Err(diesel::ConnectionError::BadConnection(e.to_string())))
+            .map(|c| SyncConnectionWrapper::new(c))
+    }
+
+    fn load<'conn, 'query, T>(&'conn mut self, source: T) -> Self::LoadFuture<'conn, 'query>
+    where
+        T: AsQuery + 'query,
+        T::Query: QueryFragment<Self::Backend> + QueryId + 'query,
+    {
+        self.execute_with_prepared_query(source.as_query(), |conn, query| {
+            use diesel::row::IntoOwnedRow;
+            conn.load(&query).map(|c| {
+                c.map(|row| row.map(IntoOwnedRow::into_owned))
+                    .collect::<Vec<QueryResult<O>>>()
+            })
+        })
+        .map_ok(|rows| futures_util::stream::iter(rows).boxed())
+        .boxed()
+    }
+
+    fn execute_returning_count<'conn, 'query, T>(
+        &'conn mut self,
+        source: T,
+    ) -> Self::ExecuteFuture<'conn, 'query>
+    where
+        T: QueryFragment<Self::Backend> + QueryId,
+    {
+        self.execute_with_prepared_query(source, |conn, query| conn.execute_returning_count(&query))
+    }
+
+    fn transaction_state(
+        &mut self,
+    ) -> &mut <Self::TransactionManager as TransactionManager<Self>>::TransactionStateData {
+        self.exclusive_connection().transaction_state()
+    }
+}
+
+/// A wrapper of a diesel transaction manager usable in async context.
+pub struct SyncTransactionManagerWrapper<T>(PhantomData<T>);
+
+#[async_trait::async_trait]
+impl<T, C> TransactionManager<SyncConnectionWrapper<C>> for SyncTransactionManagerWrapper<T>
+where
+    SyncConnectionWrapper<C>: AsyncConnection,
+    C: Connection + 'static,
+    T: diesel::connection::TransactionManager<C> + Send,
+{
+    type TransactionStateData = T::TransactionStateData;
+
+    async fn begin_transaction(conn: &mut SyncConnectionWrapper<C>) -> QueryResult<()> {
+        conn.spawn_blocking(move |inner| T::begin_transaction(inner))
+            .await
+    }
+
+    async fn commit_transaction(conn: &mut SyncConnectionWrapper<C>) -> QueryResult<()> {
+        conn.spawn_blocking(move |inner| T::commit_transaction(inner))
+            .await
+    }
+
+    async fn rollback_transaction(conn: &mut SyncConnectionWrapper<C>) -> QueryResult<()> {
+        conn.spawn_blocking(move |inner| T::rollback_transaction(inner))
+            .await
+    }
+
+    fn transaction_manager_status_mut(
+        conn: &mut SyncConnectionWrapper<C>,
+    ) -> &mut TransactionManagerStatus {
+        T::transaction_manager_status_mut(conn.exclusive_connection())
+    }
+}
+
+impl<C> SyncConnectionWrapper<C> {
+    /// Builds a wrapper with this underlying sync connection
+    pub fn new(connection: C) -> Self
+    where
+        C: Connection,
+    {
+        SyncConnectionWrapper {
+            inner: Arc::new(Mutex::new(connection)),
+        }
+    }
+
+    pub(self) fn spawn_blocking<'a, R>(
```

Comment:
Mostly a note for myself: I want to abstract that later over different `spawn_blocking` functions, so that you could use different runtimes here.

(That's nothing that needs to be addressed in this PR)

Reply:
That would be a task I am interested in after this PR!

Reply:
That would be great :pray: